src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/DocumentBuilder.cs

```diff
@@ -0,0 +1,533 @@
+                        if (modifiedExprNode != null)
+                        {
+                            var trailingTrivia = modifiedExprNode.GetTrailingTrivia();
+                            modifiedExprNode = modifiedExprNode.WithTrailingTrivia(trailingTrivia);
+                            modifiedBlockSyntaxNode = modifiedBlockSyntaxNode.ReplaceNode(parentNode, modifiedExprNode);
+                        }
+                    }
+                    //add within Lambda block of parent node.
+                    else if ((change.Type?.Equals(CodeChangeType.InLambdaBlock)).GetValueOrDefault(false))
+                    {
+                        var modifiedExprNode = AddExpressionInLambdaBlock(exprNode, change.InsertAfter, change.Block, parameterValues);
+                        if (modifiedExprNode != null)
+                        {
+                            modifiedBlockSyntaxNode = modifiedBlockSyntaxNode.ReplaceNode(parentNode, modifiedExprNode);
+                        }
+                    }
+                }
+            }
+            return modifiedBlockSyntaxNode;
+        }
+
+        internal ExpressionStatementSyntax? AddSimpleMemberAccessExpression(ExpressionStatementSyntax expression, string? codeSnippet)
+        {
+            if (!string.IsNullOrEmpty(codeSnippet) && !expression.ToString().Trim(CodeSnippetTrimChars).Contains(codeSnippet.Trim(CodeSnippetTrimChars)))
+            {
+                return expression.WithExpression(SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, expression.Expression, SyntaxFactory.IdentifierName(codeSnippet)))
+                    .WithTrailingTrivia(new SyntaxTriviaList(SemiColonTrivia));
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        internal ExpressionStatementSyntax? AddExpressionInLambdaBlock(
+            ExpressionStatementSyntax expression,
+            string? insertAfterBlock,
+            string? codeSnippet,
+            IDictionary<string, string> parameterValues)
+        {
+            BlockSyntax? blockToEdit;
+            if (!string.IsNullOrEmpty(insertAfterBlock))
+            {
+                string insertAfterFormattedBlock = FormatCodeBlock(insertAfterBlock, parameterValues);
+                blockToEdit = expression.DescendantNodes().FirstOrDefault(node =>
+                                node is BlockSyntax &&
+                                node.ToString().Trim(CodeSnippetTrimChars).Contains(insertAfterBlock)) as BlockSyntax;
+            }
+            else
+            {
+                blockToEdit = expression.DescendantNodes()
+                    .Where(node => node is BlockSyntax)
+                    .FirstOrDefault() as BlockSyntax;
+            }
+
+            if (blockToEdit != null && !string.IsNullOrEmpty(codeSnippet))
```

Comment:
>!string.IsNullOrEmpty(codeSnippet) [](http://example.com/codeflow?start=39&length=34)

if this true all previous work not needed - move on top

Reply:
fixed and moved to https://github.com/dotnet/Scaffolding/pull/1593/